xapian-core/api/valuerangeproc.cc

```diff
@@ -439,4 +439,100 @@ NumberRangeProcessor::operator()(const string& b, const string& e)
     return Xapian::Query(Xapian::Query::OP_INVALID);
 }
 
+static const int size_map[26] = {
+    0,
+    1, // B
+    0, 0, 0, 0,
+    1024 * 1024 * 1024, // G
+    0, 0, 0,
+    1024, // K
+    0,
+    1024 * 1024, // M
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+};
+
+static int
+get_bytes(char ch)
```

Comment:
This isn't a terrific name for this function; maybe `get_size_multiplier_from_suffix`? Or maybe that's too wordy.

Additionally, it should be `const char`, and `unsigned int` as the return type.